docs/guide/writing-tasks.md

```diff
@@ -649,6 +649,22 @@ task:
 
 You'll be able to manually trigger such paused tasks via Cirrus CI Web UI or directly from GitHub Checks page.
 
+## Task Execution Lock
+
+Some CI tasks perform external operations which are required to be executed one at a time. For example, parallel deploys
+to the same environment is usually a bad idea. In order to restrict parallel execution of a certain task within a repository
+you can use `execution_lock` to specify a lock key that will be used to make sure that tasks with the same lock key 
```

Comment:
```suggestion
you can use `execution_lock` to specify a lock key, a unique string that will be used to make sure that tasks with the same `execution_lock` string
```

Reply:
That is an awesome clarification! Thanks!